plugins/module_utils/network/sonic/config/lldp_interfaces/lldp_interfaces.py

```diff
@@ -362,16 +454,59 @@ def get_delete_specific_lldp_interfaces_param_requests(self, command, config):
                 if command['tlv_set'].get('ipv6_management_address') is not None:
                     url = self.lldp_intf_config_path['ipv6_management_address'].format(intf_name=name)
                     requests.append({'path': url, 'method': DELETE})
-            if 'tlv_select' in command and command['tlv_select'] is not None:
-                tlv = command['tlv_select'].get('power_management')
-                if tlv:
-                    payload = {"openconfig-lldp-ext:suppress-tlv-advertisement": ["MDI_POWER"]}
-                    url = self.lldp_intf_config_path['suppress_tlv'].format(intf_name=name)
-                    requests.append({'path': url, 'method': PATCH, 'data': payload})
-                else:
-                    url = self.lldp_intf_config_path['suppress_tlv_delete'].format(intf_name=name, med_tlv_select="MDI_POWER")
+            if 'vlan_name_tlv' in command and command['vlan_name_tlv'] is not None:
+                if command['vlan_name_tlv'].get('allowed_vlans') is not None:
```

Comment:
Why isn't deletion of a list of individual vlans and ranges of vlans being supported here? (SONiC nominally supports both of these.)

Reply:
modified the function handle individual vlan deletion as well.

Reply:
The modified logic is a step in the right direction, but due to a SONIC bug, it doesn't work for ranges. I have posted a separate comment for this problem, so I will mark this issue "Resolved".